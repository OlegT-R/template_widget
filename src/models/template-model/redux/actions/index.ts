import { ITemplate } from '../../entities';

export const UPDATE_TEMPLATE = 'template/UPDATE';

export interface IUpdateTemplateAction {
  type: string;
  payload: {
    id: string;
    template: ITemplate;
  };
}

export const updateTemplate = (id: string, template: ITemplate) => ({
  type: UPDATE_TEMPLATE,
  payload: {
    id,
    template,
  },
});
