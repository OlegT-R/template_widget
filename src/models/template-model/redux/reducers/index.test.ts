import { templates } from '.';
import { UPDATE_TEMPLATE } from '../actions';

const template = {
  block: [{
    id: 'textArea1',
    value: 'test'
  }]
};

describe('templates reducers', () => {
  test('template_reducer/UPDATE', () => {
    const action = {
      type: UPDATE_TEMPLATE,
      payload: {
        id: 'testId',
        template,
      },
    };

    const newState = templates({}, action);
    expect(newState.testId.template).toBe(template);
    expect(newState).toMatchSnapshot();
  });
});
