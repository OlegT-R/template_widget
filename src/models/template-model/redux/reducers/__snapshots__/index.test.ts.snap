// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates reducers template_reducer/UPDATE 1`] = `
Object {
  "testId": Object {
    "template": Object {
      "block": Array [
        Object {
          "id": "textArea1",
          "value": "test",
        },
      ],
    },
  },
}
`;
